### PR TITLE
feat: `useMenu` hook in `@pankod/refine-core`

### PR DESCRIPTION
**Pull request by aliemir**
Move `useMenu` hook from UI packages to `@pankod/refine-core`.

- [x] added `useMenu` hook to `@pankod/refine-core`
- [x] deprecated `useMenu` hooks from `@pankod/refine-mui` and `@pankod/refine-antd`
- [x] replaced deprecated `useMenu` usage with the one from `@pankod/refine-core`
- [x] initialize `useMenu` documentation for `@pankod/refine-core`
- [x] update `useResources` menu from documentations with `useMenu`
- [x] update `useMenu` documentation for `@pankod/refine-core`
- [x] add deprecate sign to `useMenu` in `@pankod/refine-antd` and `@pankod/refine-mui`
- [x] move **Custom Sider** examples from `useMenu` docs to `UI Frameworks > * > Customization`
- [x] move **Custom Layout** examples from `useMenu` docs to `UI Frameworks > * > Customization`
- [x] move **Custom Theme** examples from `useMenu` docs to `UI Frameworks > * > Customization`
- [x] add custom route support to `defaultOpenKeys`

### CODE REVIEW

File: documentation/docs/core/hooks/ui/useMenu.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+:::
+
+## Usage
+
+:::tip
+
+If you are using [`@pankod/refine-antd`](#) or [`@pankod/refine-mui`](#) as a UI framework integration, you can find out more info about how their `<Sider/>` components are created and how to create a custom one by following their guides.
+
+[Ant Design > Customization > Custom Sider&#8594](#)
+[Material UI > Customization > Custom Sider&#8594](#)
```
Comment (omeraplak) [inline documentation/docs/core/hooks/ui/useMenu.md:29]:
```suggestion
[Ant Design > Customization > Custom Sider&#8594](#)

[Material UI > Customization > Custom Sider&#8594](#)
```

File: documentation/docs/core/hooks/ui/useMenu.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+:::
+
+### Creating a Menu
+
+We will show you how to use `useMenu` to create a simple menu for your refine application.
+
+Create a `<Layout />` component inside `src/components/Layout.tsx` with the following code;
+
+```tsx title="components/Layout.tsx"
```
Comment (omeraplak) [inline documentation/docs/core/hooks/ui/useMenu.md:38]:
```suggestion
```tsx title="components/layout.tsx"
```

File: documentation/docs/core/hooks/ui/useMenu.md [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+                    show: PostShow,
+                },
+                {
+                    name: "categories",
+                    list: CategoryList,
+                    create: CategoryCreate,
+                    edit: CategoryEdit,
+                    canDelete: true,
+                },
+            ]}
```
Comment (salihozdemir) [inline documentation/docs/core/hooks/ui/useMenu.md:129]: I think, resource pages should have import path.

File: documentation/docs/core/hooks/ui/useMenu.md [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+            // highlight-end
+            Layout={Layout}
+        />
+    );
+};
+```
+
+Now you can update your `<Layout/>` to support multi level rendering with following code:
+
+```tsx title="components/Layout.tsx"
```
Comment (salihozdemir) [inline documentation/docs/core/hooks/ui/useMenu.md:187]:
```suggestion
```tsx title="src/components/Layout.tsx"
```

File: documentation/docs/ui-frameworks/antd/customization/layout.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+                    <AntdLayout.Content>
+                        <AntdLayout.Content>
+                            <div style={{ padding: 24, minHeight: 360 }}>
+                                {children}
+                            </div>
+                        </AntdLayout.Content>
+                        {Footer && <Footer />}
+                    </AntdLayout.Content>
+                    {OffLayoutArea && <OffLayoutArea />}
+                </AntdLayout>
```
Comment (salihozdemir) [inline documentation/docs/ui-frameworks/antd/customization/layout.md:62]:
```suggestion
            Layout={({ children }) => (
                <AntdLayout>
                    <AntdLayout.Header>
                        <CustomSider />
                    </AntdLayout.Header>
                    <AntdLayout.Content>
                        <AntdLayout.Content>
                            <div style={{ padding: 24, minHeight: 360 }}>
                                {children}
                            </div>
                        </AntdLayout.Content>
                    </AntdLayout.Content>
                </AntdLayout>
```

By default, `Footer` and `OffLayoutArea` return null. I think, we can remove them for no confusing.

File: examples/customization/customTheme/mui/public/index.html [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+        <title>Material UI base example</title>
```
Comment (salihozdemir) [inline examples/customization/customTheme/mui/public/index.html:26]:
```suggestion
        <title>Material UI custom theme example</title>
```

File: examples/customization/customTheme/mui/public/manifest.json [hunk 7]
```diff
@@ -0,0 +1,15 @@
+{
+    "short_name": "Material UI base example",
+    "name": "Material UI base example",
```
Comment (salihozdemir) [inline examples/customization/customTheme/mui/public/manifest.json:3]:
```suggestion
    "short_name": "Material UI custom theme example",
    "name": "Material UI custom theme example",
```

File: examples/customization/customTheme/mui/src/pages/posts/create.tsx [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+                    fullWidth
+                    label="Title"
+                    name="title"
+                    autoFocus
+                />
+                <Controller
+                    control={control}
+                    name="status"
+                    rules={{ required: "This field is required" }}
+                    defaultValue=""
```
Comment (salihozdemir) [inline examples/customization/customTheme/mui/src/pages/posts/create.tsx:47]:
```suggestion
                   
```

File: examples/customization/customTheme/mui/src/pages/posts/edit.tsx [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+                    fullWidth
+                    label="Title"
+                    name="title"
+                    autoFocus
+                />
+                <Controller
+                    control={control}
+                    name="status"
+                    rules={{ required: "This field is required" }}
+                    defaultValue=""
```
Comment (salihozdemir) [inline examples/customization/customTheme/mui/src/pages/posts/edit.tsx:49]:
```suggestion
                    defaultValue={null}
```

**Comment by aliemir** [comment]
Updated live examples from CodeSandbox to StackBlitz as in #1981

File: documentation/docs/core/hooks/ui/useMenu.md (hunk 3 above)
Comment (aliemir) [reply documentation/docs/core/hooks/ui/useMenu.md:131]: Added imports to the top 👍

File: documentation/docs/ui-frameworks/antd/customization/layout.md (hunk 5 above)
Comment (aliemir) [reply documentation/docs/ui-frameworks/antd/customization/layout.md:62]: Even if they are null, the aim is to replicate the default layout in the custom layout example to give a better starting point to customize for users. By removing those in a custom layout, we will lose the functionality on `OffLayoutArea` and `Footer` props in `<Refine/>` component.

File: documentation/docs/core/hooks/ui/useMenu.md [hunk 10]
```diff
@@ -0,0 +1,289 @@
...
+
+`useMenu` hook comes out of the box with multi level menu support, you can render menu items recursively to get a multi level menu.
+
+Update your `resources` in `<Refine/>` with `parentName` to nest them inside a label.
+
+```tsx title="src/App.tsx"
+import { Refine } from "@pankod/refine-core";
+import routerProvider from "@pankod/refine-react-router-v6";
+import dataProvider from "@pankod/refine-simple-rest";
+
```
Comment (salihozdemir) [inline documentation/docs/core/hooks/ui/useMenu.md:149]:
```suggestion

import { PostList, PostCreate, PostEdit, PostShow } from "pages/post";
import { CategoryList, CategoryCreate, CategoryEdit } from "pages/category";
```

File: documentation/docs/core/hooks/ui/useMenu.md [hunk 11]
```diff
@@ -0,0 +1,289 @@
+---
+id: useMenu
+title: useMenu
+---
+
+`useMenu` is used to get menu items derived from the resources. These items include a link to dashboard page (if it exists) and links to the user defined resources (passed as children to `<Refine>`).
+This hook can also be used to build custom menus, including multi-level support. `<Sider/>` components inside [`@pankod/refine-antd`](#) and [`@pankod/refine-mui`](#) packages are using this hook as a base for their menus.
```
Comment (salihozdemir) [inline documentation/docs/core/hooks/ui/useMenu.md:7]: Missing links

File: documentation/docs/core/hooks/ui/useMenu.md [hunk 12]
```diff
@@ -0,0 +1,289 @@
...
+
+## Usage
+
+:::tip
+
+If you are using [`@pankod/refine-antd`](#) or [`@pankod/refine-mui`](#) as a UI framework integration, you can find out more info about how their `<Sider/>` components are created and how to create a custom one by following their guides.
+
+[Ant Design > Customization > Custom Sider &#8594](#)
+
+[Material UI > Customization > Custom Sider &#8594](#)
```
Comment (salihozdemir) [inline documentation/docs/core/hooks/ui/useMenu.md:32]: Missing links

**Review by salihozdemir — APPROVED** [review]

**Review by ozkalai — APPROVED** [review]

File: documentation/docs/core/hooks/ui/useMenu.md [hunk 13]
```diff
@@ -0,0 +1,292 @@
...
+
+[Material UI > Customization > Custom Sider &#8594](/docs/ui-frameworks/mui/customization/sider/)
+
+:::
+
+### Creating a Menu
+
+We will show you how to use `useMenu` to create a simple menu for your refine application.
+
+Create a `<Layout />` component inside `src/components/Layout.tsx` with the following code;
```
Comment (omeraplak) [inline documentation/docs/core/hooks/ui/useMenu.md:40]:
```suggestion
Create a `<Layout />` component inside `src/components/layout.tsx` with the following code;
```

File: documentation/docs/core/hooks/ui/useMenu.md [hunk 14]
```diff
@@ -0,0 +1,292 @@
...
+                    </div>
+                </div>
+            </div>
+            <div className="bg-white">{children}</div>
+        </div>
+    );
+};
+```
+
+We created a header with a logo and a list of links to all menu items (resources). The links are clickable and will navigate to the corresponding resource. To do this, we used the [`useMenu`](/core/hooks/ui/useMenu.md) hook to get the menu items from the `<Refine/>` and the [`useRouterContext`](#) hook to get the `<Link/>` component from the router provider. Also [`useNavigation`](/core/hooks/navigation/useNavigation.md) hook can be used to navigate between routes.
```
Comment (omeraplak) [inline documentation/docs/core/hooks/ui/useMenu.md:89]:
```suggestion
We created a header with a logo and a list of links to all menu items (resources). The links are clickable and will navigate to the corresponding resource. To do this, we used the [`useMenu`](/core/hooks/ui/useMenu.md) hook to get the menu items from the `<Refine/>` and the `useRouterContext` hook to get the `<Link/>` component from the router provider. Also [`useNavigation`](/core/hooks/navigation/useNavigation.md) hook can be used to navigate between routes.
```

File: documentation/docs/ui-frameworks/mui/customization/sider.md [hunk 15]
```diff
@@ -1,6 +1,6 @@
 ---
-id: useMenu
-title: useMenu
+id: sider
```
Comment (omeraplak) [inline documentation/docs/ui-frameworks/mui/customization/sider.md:2]:
```suggestion
id: mui-sider
```
must be unique

File: documentation/docs/ui-frameworks/antd/customization/sider.md [hunk 16]
```diff
@@ -1,45 +1,33 @@
 ---
-id: useMenu
-title: useMenu
+id: sider
```
Comment (omeraplak) [inline documentation/docs/ui-frameworks/antd/customization/sider.md:2]:
```suggestion
id: antd-sider
```

**Review by omeraplak — CHANGES_REQUESTED** [review]

**Review by omeraplak — APPROVED** [review]

**Pull request merged**